realm/src/main/java/io/realm/DynamicRealm.java

```diff
@@ -0,0 +1,277 @@
+/*
+ * Copyright 2015 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.realm;
+
+import android.os.Looper;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CopyOnWriteArrayList;
+
+import io.realm.exceptions.RealmException;
+import io.realm.internal.Table;
+import io.realm.internal.TableView;
+
+/**
+ * DynamicRealm is a dynamic variant of {@link io.realm.Realm}. This means that all access to data and/or queries are
+ * done using string based class names instead of class type references.
+ *
+ * The same {@link io.realm.RealmConfiguration} can be used to open a Realm file in both dynamic and typed mode, but
+ * modifying the schema while doing so is highly discouraged and will most likely crash the typed Realm.
```

Comment:
this is confusing, we'll get a DynamicRealm for migration (this will modify the schema)?